osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
+                        print(" " + itinerary_stop.name +
+                              " - " + itinerary_stop.osm_id)
+                        break
+                    gtfs_trip.AddStopTime(gtfs_stop, stop_time=time_at_stop)
+
+                # Add stop without time information, too (we interpolate later)
+                else:
+                    try:
+                        gtfs_trip.AddStopTime(gtfs_stop)
+                    except ValueError:
+                        print("Warning: Could not add a stop to trip.")
+                        print(" " + itinerary_stop.name +
+                              " - " + itinerary_stop.osm_id)
+                        break
+
+                # Add reference to shape
+                gtfs_trip.shape_id = shape_id
+
+                # Add empty attributes to make navitia happy
+                gtfs_trip.block_id = ""
+                gtfs_trip.wheelchair_accessible = ""
+                gtfs_trip.bikes_allowed = ""
+                gtfs_trip.direction_id = ""
+
+            # Calculate all times of stops, which were added with no time
+            Helper.interpolate_stop_times(gtfs_trip)
+        return trips_count
+
+    def _create_gtfs_service_period(self, feed, service):
+        """
+        Generate a transitfeed ServicePeriod object
+        from a time string according to the standard schedule:
+        https://github.com/grote/osm2gtfs/wiki/Schedule
+
+        :return: ServicePeriod object
+        """
+        try:
+            gtfs_service = feed.GetServicePeriod(service)
+            if gtfs_service is not None:
+                return gtfs_service
+        except KeyError:
+            pass
+
+        if service == "Mo-Fr":
+            gtfs_service = ServicePeriod(service)
+            gtfs_service.SetWeekdayService(True)
+            gtfs_service.SetWeekendService(False)
+        elif service == "Mo-Sa":
+            gtfs_service = ServicePeriod(service)
+            gtfs_service.SetWeekdayService(True)
+            gtfs_service.SetWeekendService(False)
+            gtfs_service.SetDayOfWeekHasService(5, True)
+        elif service == "Mo-Su":
+            gtfs_service = ServicePeriod(service)
+            gtfs_service.SetWeekdayService(True)
+            gtfs_service.SetWeekendService(True)
+        elif service == "Sa":
+            gtfs_service = ServicePeriod(service)
+            gtfs_service.SetWeekdayService(False)
+            gtfs_service.SetWeekendService(False)
+            gtfs_service.SetDayOfWeekHasService(5, True)
+        elif service == "Su":
+            gtfs_service = ServicePeriod(service)
+            gtfs_service.SetWeekdayService(False)
+            gtfs_service.SetWeekendService(False)
+            gtfs_service.SetDayOfWeekHasService(6, True)
+        elif service == "Sa-Su":
+            gtfs_service = ServicePeriod(service)
+            gtfs_service.SetWeekdayService(False)
+            gtfs_service.SetWeekendService(True)
```

Comment:
I think this should be done with another better automatic algorithm. But I think it is fine for now.

Reply:
Split out to #111.

Reply:
> Split out to #111.

I'm sorry, wrong comment.